### PR TITLE
Removed Span<T> and ReadOnlySpan<T> form System.Buffers

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
We will now be using the Span<T> defined in System.Slices
I converted all usages of ReadOnlySpan<T> to Span<T>. Once we add ReadOnlySpan<T>
to System.Slices, I will convert the formatting code back. I left some TODOs there.

Also, this change resulted in a 10-20% regression in formatting perf. I am pretty sure it's
due to unoptimized Span<T>.Set methods (extensions). There is TODO in the code (see TryCopyTo) to optimize it.

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
@krwq, @sokket, could you take a look?

File: src/System.Text.Formatting.Globalization/src/project.lock.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "locked": true,
+  "locked": false,
```
Comment (sokket) [inline src/System.Text.Formatting.Globalization/src/project.lock.json]: I thought we wanted to keep this true...is that not correct?

Comment (sokket) [inline src/System.Text.Formatting.Globalization/src/project.lock.json]: And below

**Comment by sokket** [comment]
One question but otherwise LGTM

File: src/System.Text.Formatting.Globalization/src/project.lock.json (hunk 1 above)
Comment (KrzysztofCwalina) [inline src/System.Text.Formatting.Globalization/src/project.lock.json]: I just changed the project.json file to not use wildcards, and so I think nuget changes locked to false as there is nothing to ever upgrade to.

**Pull request merged**